data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/CloudEventRequestToRecordMapper.java

```diff
@@ -0,0 +1,60 @@
+package dev.knative.eventing.kafka.broker.receiver;
+
+import static java.lang.String.join;
+
+import io.cloudevents.CloudEvent;
+import io.cloudevents.core.message.Message;
+import io.cloudevents.http.vertx.VertxMessageFactory;
+import io.cloudevents.lang.Nullable;
+import io.vertx.core.Future;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.kafka.client.producer.KafkaProducerRecord;
+import java.util.StringTokenizer;
+
+public class CloudEventRequestToRecordMapper implements RequestToRecordMapper<String, CloudEvent> {
+
+  static final int PATH_TOKEN_NUMBER = 2;
+  static final String PATH_DELIMITER = "/";
+  static final String TOPIC_DELIMITER = "-";
+  public static final String TOPIC_PREFIX = "knative-";
+
+  @Override
+  public Future<KafkaProducerRecord<String, CloudEvent>> recordFromRequest(
+      final HttpServerRequest request) {
+
+    return VertxMessageFactory.fromHttpServerRequest(request)
+        // TODO is this conversion really necessary?
```

Comment:
I would say that we could avoid to convert to event but let's first complete the broker this way, then let's work to optimize it